patches/minecraft/net/minecraft/block/BlockChorusFlower.java.patch

```diff
@@ -0,0 +1,204 @@
+-                            {
+-                                if (block1 == Blocks.field_150377_bs)
+-                                {
+-                                    flag1 = true;
+-                                }
+-
+-                                break;
+-                            }
+-
+-                            ++j;
+-                        }
+-
+-                        int i1 = 4;
+-
+-                        if (flag1)
+-                        {
+-                            ++i1;
+-                        }
+-
+-                        if (j < 2 || p_180650_4_.nextInt(i1) >= j)
+-                        {
+-                            flag = true;
+-                        }
+-                    }
+-                    else if (iblockstate.func_185904_a() == Material.field_151579_a)
+-                    {
+-                        flag = true;
+-                    }
+-
+-                    if (flag && func_185604_a(p_180650_1_, blockpos, (EnumFacing)null) && p_180650_1_.func_175623_d(p_180650_2_.func_177981_b(2)))
+-                    {
+-                        p_180650_1_.func_180501_a(p_180650_2_, Blocks.field_185765_cR.func_176223_P(), 2);
+-                        this.func_185602_a(p_180650_1_, blockpos, i);
+-                    }
+-                    else if (i < 4)
+-                    {
+-                        int l = p_180650_4_.nextInt(4);
+-                        boolean flag2 = false;
+-
+-                        if (flag1)
+-                        {
+-                            ++l;
+-                        }
+-
+-                        for (int j1 = 0; j1 < l; ++j1)
+-                        {
+-                            EnumFacing enumfacing = EnumFacing.Plane.HORIZONTAL.func_179518_a(p_180650_4_);
+-                            BlockPos blockpos1 = p_180650_2_.func_177972_a(enumfacing);
+-
+-                            if (p_180650_1_.func_175623_d(blockpos1) && p_180650_1_.func_175623_d(blockpos1.func_177977_b()) && func_185604_a(p_180650_1_, blockpos1, enumfacing.func_176734_d()))
+-                            {
+-                                this.func_185602_a(p_180650_1_, blockpos1, i + 1);
+-                                flag2 = true;
+-                            }
+-                        }
+-
+-                        if (flag2)
+-                        {
+-                            p_180650_1_.func_180501_a(p_180650_2_, Blocks.field_185765_cR.func_176223_P(), 2);
+-                        }
+-                        else
+-                        {
+-                            this.func_185605_c(p_180650_1_, p_180650_2_);
+-                        }
+-                    }
+-                    else if (i == 4)
+-                    {
+-                        this.func_185605_c(p_180650_1_, p_180650_2_);
+-                    }
++                    BlockEvent ev = new BlockEvent.CropGrowEvent.Pre(p_180650_1_,p_180650_2_,p_180650_3_);
```

Comment:
Ok thank you. I was trying to follow the "indent your code properly" convention as I saw nothing else about it (and still don't).